atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransactionManager.java

```diff
@@ -134,21 +134,22 @@ public SerializableTransactionManager(KeyValueService keyValueService,
             int concurrentGetRangesThreadPoolSize,
             int defaultGetRangesConcurrency,
             long timestampCacheSize) {
-        this(keyValueService,
-                timestampService,
-                lockClient,
+        return new SerializableTransactionManager(keyValueService,
+                new LegacyTimelockService(timestampService, lockService, lockClient),
                 lockService,
                 transactionService,
                 constraintModeSupplier,
                 conflictDetectionManager,
                 sweepStrategyManager,
                 cleaner,
                 false,
+                () -> AtlasDbConstants.DEFAULT_TRANSACTION_LOCK_ACQUIRE_TIMEOUT_MS,
                 concurrentGetRangesThreadPoolSize,
                 defaultGetRangesConcurrency,
                 timestampCacheSize);
     }
 
+    // Used by internal product.
```

Comment:
Do we intend to deprecate this, get the internal product to remove the usage, and eventually remove this constructor? If so, let's `@Deprecate` this one.

Reply:
Ah, checkstyle's going to be mad at you now, because you didn't add a javadoc explaining why it's deprecated.